packages/cicero-core/test/templatelibrary.js

```diff
@@ -45,6 +70,17 @@ describe('TemplateLibrary', () => {
             templateIndex.should.have.property('helloworld@0.3.0');
         });
 
+        it('should retrieve index for template library without authentication', async () => {
+            mock('rp', {});
+            TemplateLibrary = mock.reRequire('../lib/templatelibrary');
+            const templateLibrary = new TemplateLibrary();
+            rp.get('https://templates.accordproject.org')
+                .then(async () => {
+                    const templateIndex = await templateLibrary.getTemplateIndex();
+                    templateIndex.should.have.property('helloworld@0.3.0');
+                });
+        });
+
```

Comment:
@mttrbrts I mocked in this fashion. Is this fine or I have to make other changes. As well whenever I run 
`templateLibrary.httpHeader.should.equal(null)`
It gives an error. Need a little assist of yours for this .

Reply:
Ah, I see the issue. We should be mocking `request-promise`, rather than `rp`, and we don't need to mock `'../lib/templateLibrary'`. We want to test the behaviour of `TemplateLibrary`, but to make it independent of any network requests. 

For example, something like this ...
```
it('should retrieve index for template library without authentication', async () => {
            let getParams = {};
            mock('request-promise', { get: (params) => { getParams = params; } });
            const templateLibrary = new TemplateLibrary();
            await templateLibrary.getTemplateIndex();
           getParams.headers.authorization.should.equal(undefined);
        });
```
Note, please treat this are pseudo-code; it is unlikely to work exactly as is.

Reply:
@mttrbrts Sure I will thanks for the pseudo code.